checkov/policies3d/runner.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import annotations
+
+import logging
+from enum import Enum
+
+from checkov.common.bridgecrew.check_type import CheckType
+from checkov.common.models.enums import CheckResult
+from checkov.common.output.record import Record
+from checkov.common.output.report import Report
+
+from checkov.common.runners.base_post_runner import BasePostRunner
+from checkov.common.util.type_forcers import force_list
+from checkov.policies3d.checks_infra.base_check import Base3dPolicyCheck
+from checkov.runner_filter import RunnerFilter
+
+
+class CVEAttribute(str, Enum):
+    RISK_FACTORS = "risk_factors"
+
+
+class Policy3dRunner(BasePostRunner):
+    check_type = CheckType.POLICY_3D  # noqa: CCE003  # a static attribute
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def run(  # type:ignore[override]
+            self,
+            checks: list[Base3dPolicyCheck] | None = None,
+            scan_reports: list[Report] | None = None,
+            runner_filter: RunnerFilter | None = None
+    ) -> Report | list[Report]:
+        runner_filter = runner_filter or RunnerFilter()
+        if not runner_filter.show_progress_bar:
+            self.pbar.turn_off_progress_bar()
+
+        report = Report(self.check_type)
+
+        if not checks or not scan_reports:
+            logging.debug("No checks or reports scan.")
+            return report
+
+        self.pbar.initiate(len(checks))
+
+        reports_by_fw = {report.check_type: report for report in scan_reports}
+        for check in checks:
+            records = self.run_check(check, reports_by_fw)
+            for record in records:
+                report.add_record(record=record)
+
+        self.pbar.close()
+        return report
+
+    def run_check(self, check: Base3dPolicyCheck, reports_by_fw: dict[str, Report]) -> list[Record]:
+        records = []
+        iac_results_map = self.solve_check_iac(check, reports_by_fw)
+        cve_results_map = self.solve_check_cve(check, reports_by_fw)
+
+        for iac_resource, iac_records in iac_results_map.items():
+            for cve_resource in cve_results_map:
+                if iac_resource == cve_resource:
+                    # This means we found the combination on the same resource -> create a violation for that resource
+                    check_result = CheckResult.FAILED
+                    record = self.get_record(check, iac_records[0], check_result)
```

Comment:
```suggestion
                    record = self.get_record(check, iac_resource, check_result)
```

Reply:
Notice your change swaps a record type for a string type, and I need the record here..

Reply:
Is it guaranteed that `iac_records[0]` is the matching record for the resource? Shouldn't we match it?

Reply:
We need the iac record just for some attributes we insert to our own 3d policy record, and these attributes should be the same in all of the records for the same resource, so I just take the first one